consensus/transaction_solidifier/transaction_solidifier.c

```diff
@@ -292,25 +299,29 @@ static retcode_t check_transaction_and_update_solid_state(
   }
 
   if (!transaction_solid(transaction)) {
-    *is_new_solid = true;
-
     if ((ret = check_approvee_solid_state(
-             ts, tangle, transaction_trunk(transaction), is_new_solid)) !=
+             ts, tangle, transaction_trunk(transaction), &is_trunk_solid)) !=
         RC_OK) {
       *is_new_solid = false;
+      log_error(TRANSACTION_SOLIDIFIER_LOGGER_ID,
+                "Checking solidity of trunk failed\n");
       return ret;
     }
 
     if ((ret = check_approvee_solid_state(
-             ts, tangle, transaction_branch(transaction), is_new_solid)) !=
+             ts, tangle, transaction_branch(transaction), &is_branch_solid)) !=
```

Comment:
Instead of having to booleans, we should not query for branch solid state if trunk is not solid